zephyr/wrapper.c

```diff
@@ -64,28 +94,42 @@ void heap_trace_all(int force)
  * used by care for matching.
  */
 
-/* register an IRQ  handler - irq number needs mapped to zephyr IRQ */
+static uint32_t to_zephyr_irq(uint32_t sof_irq) {
+        return SOC_AGGREGATE_IRQ(SOF_IRQ_BIT(sof_irq),
+                                 SOF_IRQ_NUMBER(sof_irq));
+}
+
 int interrupt_register(uint32_t irq, void(*handler)(void *arg), void *arg)
 {
-	return 0;
+	uint32_t zephyr_irq = to_zephyr_irq(irq);
+
+	return arch_irq_connect_dynamic(zephyr_irq, 0, handler, arg, 0);
 }
 
 /* unregister an IRQ handler - matches on IRQ number and data ptr */
 void interrupt_unregister(uint32_t irq, const void *arg)
 {
+	/*
+	 * There is no "unregister" (or "disconnect") for
+         * interrupts in Zephyr.
+         */
```

Comment:
This may be a problem in the future, can't we just register a dummy handler? Assuming Zephyr doesn't support shared interrupts either.

Reply:
We can use statically mapped (zephyr)  IRQs in the future. I don't think we have a use case where we need to unload a driver today to save memory.

Reply:
I guess I should familiarize myself with Zephyr.